src/lxc/network.c

```diff
@@ -302,8 +302,10 @@ static int instantiate_veth(struct lxc_handler *handler, struct lxc_netdev *netd
 	if (!veth2)
 		return -1;
 
-	if (netdev->mtu && lxc_safe_uint(netdev->mtu, &mtu)) {
-		return log_error_errno(-1, errno, "Failed to parse mtu");
+	/* if mtu is specified in config then use that, otherwise inherit from link device if provided. */
+	if (netdev->mtu) {
+		if (lxc_safe_uint(netdev->mtu, &mtu))
+			return log_error_errno(-1, errno, "Failed to parse mtu");
```

Comment:
clang's not happy :)
```
../../../src/lxc/monitor.c  -fPIC -DPIC -o .libs/liblxc_la-monitor.o
../../../src/lxc/network.c:313:7: error: variable 'mtu' is used uninitialized
      whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
                if (ifindex_mtu) {
                    ^~~~~~~~~~~
../../../src/lxc/network.c:319:52: note: uninitialized use occurs here
        err = lxc_veth_create(veth1, veth2, handler->pid, mtu);
                                                          ^~~
../../../src/lxc/network.c:313:3: note: remove the 'if' if its condition is
      always true
                if (ifindex_mtu) {
                ^~~~~~~~~~~~~~~~~
../../../src/lxc/network.c:309:13: error: variable 'mtu' is used uninitialized
      whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
        } else if (netdev->link[0] != '\0') {
                   ^~~~~~~~~~~~~~~~~~~~~~~
../../../src/lxc/network.c:319:52: note: uninitialized use occurs here
        err = lxc_veth_create(veth1, veth2, handler->pid, mtu);
                                                          ^~~
../../../src/lxc/network.c:309:9: note: remove the 'if' if its condition is
      always true
        } else if (netdev->link[0] != '\0') {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../src/lxc/network.c:276:18: note: initialize the variable 'mtu' to
      silence this warning
        unsigned int mtu;
                        ^
                         = 0
2 errors generated.
Makefile:2021: recipe for target 'liblxc_la-network.lo' failed
make[3]: *** [liblxc_la-network.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
```

Reply:
@brauner I've pushed with it initialised to 1500